src/java/map/FloatMask.java

```diff
@@ -422,6 +422,118 @@ public FloatMask setSize(int size) {
         return this;
     }
 
+    public FloatMask removeValuesOutsideOf(BinaryMask other) {
+        for (int x = 0; x < getSize(); x++) {
+            for (int y = 0; y < getSize(); y++) {
+                if (!other.get(x, y)) {
+                    set(x, y, 0f);
+                }
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeValuesInsideOf(BinaryMask other) {
+        for (int x = 0; x < getSize(); x++) {
+            for (int y = 0; y < getSize(); y++) {
+                if (other.get(x, y)) {
+                    set(x, y, 0f);
+                }
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeValuesOutsideOfRange(float min, float max) {
+        for (int y = 0; y < getSize(); y++) {
+            for (int x = 0; x < getSize(); x++) {
+                if (this.get(x, y) < min || this.get(x, y) > max) {
+                    set(x, y, 0f);
+                }
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask removeValuesInRange(float min, float max) {
+        subtract(this.copy().removeValuesOutsideOfRange(min, max));
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask replaceValuesInRangeWith(BinaryMask range, FloatMask replacement) {
+        removeValuesInsideOf(range).add(replacement.copy().removeValuesOutsideOf(range));
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask copyWithinRange(BinaryMask other) {
+        FloatMask newMask = copy().removeValuesOutsideOf(other);
+        VisualDebugger.visualizeMask(this);
+        return newMask;
```

Comment:
Is this supposed to return newMask or is it supposed to modify this mask?
Also this would break the visualizer logic if we are expecting the same mask back.
Aslo the call to visualize mask doesnt matter currently as this is never changed

Reply:
return new mask
visualize can be removed if there's no good way to do that